src/main.c

```diff
@@ -37,15 +37,15 @@ static char *read_file(const char *path, size_t *size) {
 static void dump_ast(const char *code, int binary) {
   struct ast ast;
 
-  ast_init(&ast, 0);
+  mbuf_init(&ast.mbuf, 0);
```

Comment:
Since we keep the `struct ast` (mostly for self documenting purposes, but it might be useful to keep more AST related stuff) wouldn't it make sense to keep the `ast_init` function so that we don't have to locate all invocations of `mbuf_init` that initialize a `struct ast::buf` value?